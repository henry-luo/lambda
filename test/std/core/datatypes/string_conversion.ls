// Test: String Type Conversion and Formatting
// Category: core/datatypes
// Type: positive
// Expected: 1

// String to other types
let num_str = "42"
num_str
let num = num_str as int           
num
let float_num = "3.14" as float    
float_num
let bool_true = "true" as bool     
bool_true
let bool_false = "false" as bool   
bool_false
let int_str = 42 as string         
int_str
let float_str = 3.14 as string     
float_str
let bool_str = true as string      
bool_str
let null_str = null as string      
null_str
let name = "Alice"
name
let age = 30
age
let formatted = `Name: ${name}, Age: ${age}`  
formatted
let price = 19.99
price
let price_str = price.format("0.00")  
price_str
let now = date()
now
let date_str = now.format("YYYY-MM-DD")  
date_str
let padded = "42".pad_start(5)     
padded
let right_aligned = "42".rjust(5)   
right_aligned
let left_aligned = "42".ljust(5)    
left_aligned
let centered = "42".center(6, "-")  
centered
1
