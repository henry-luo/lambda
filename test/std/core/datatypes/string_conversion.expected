09:15:42 [NOTICE] Start transpiling test/std/core/datatypes/string_conversion.ls...
09:15:42 [ERROR] Syntax tree has errors.
09:15:42 [ERROR] Syntax error at Ln 8, Col 11 - 8, Col 21: ERROR
09:15:42 [ERROR] ERROR node at Ln 8, Col 11 - 8, Col 21
09:15:42 [ERROR] Syntax error at Ln 9, Col 17 - 9, Col 26: ERROR
09:15:42 [ERROR] ERROR node at Ln 9, Col 17 - 9, Col 26
09:15:42 [ERROR] Syntax error at Ln 10, Col 17 - 10, Col 26: ERROR
09:15:42 [ERROR] ERROR node at Ln 10, Col 17 - 10, Col 26
09:15:42 [ERROR] Syntax error at Ln 11, Col 18 - 11, Col 28: ERROR
09:15:42 [ERROR] ERROR node at Ln 11, Col 18 - 11, Col 28
09:15:42 [ERROR] Syntax error at Ln 14, Col 15 - 14, Col 20: ERROR
09:15:42 [ERROR] ERROR node at Ln 14, Col 15 - 14, Col 20
09:15:42 [ERROR] Syntax error at Ln 15, Col 17 - 15, Col 24: ERROR
09:15:42 [ERROR] ERROR node at Ln 15, Col 17 - 15, Col 24
09:15:42 [ERROR] Syntax error at Ln 16, Col 16 - 16, Col 23: ERROR
09:15:42 [ERROR] ERROR node at Ln 16, Col 16 - 16, Col 23
09:15:42 [ERROR] Syntax error at Ln 17, Col 16 - 17, Col 23: ERROR
09:15:42 [ERROR] ERROR node at Ln 17, Col 16 - 17, Col 23
09:15:42 [ERROR] Syntax error at Ln 22, Col 1 - 22, Col 45: ERROR
09:15:42 [ERROR] ERROR node at Ln 22, Col 1 - 22, Col 45
09:15:42 [ERROR] Syntax error at Ln 22, Col 17 - 22, Col 25: ERROR
09:15:42 [ERROR] ERROR node at Ln 22, Col 17 - 22, Col 25
09:15:42 [ERROR] Syntax error at Ln 22, Col 17 - 22, Col 18: ERROR
09:15:42 [ERROR] ERROR node at Ln 22, Col 17 - 22, Col 18
09:15:42 [ERROR] Syntax error at Ln 22, Col 38 - 22, Col 39: ERROR
09:15:42 [ERROR] ERROR node at Ln 22, Col 38 - 22, Col 39
09:15:42 [ERROR] Syntax error at Ln 22, Col 44 - 22, Col 45: ERROR
09:15:42 [ERROR] ERROR node at Ln 22, Col 44 - 22, Col 45
09:15:42 [ERROR] Error: Failed to compile the function.
##### Script 'test/std/core/datatypes/string_conversion.ls' executed: #####
null
